src/com/cft/contactmerge/NameMatchSelector.java

```diff
@@ -0,0 +1,75 @@
+/* Modified by James Page on May 8th, 2018
+ * Contains the selections of which type of naming properties will be used for matching
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+
+public abstract class NameMatchSelector extends NameMatchLogic {
+    /************************************************************************
+     *********** Method(s) used to call specific name comparisons ***********
+     ************************************************************************/
+    public static AnswerType doFirstNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactTwoFirstName){
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+
+        return FirstNameMatcher.getFirstNameMatchResult(editedFirstNameOne, editedFirstNameTwo);
+    }
+
+    public static AnswerType doLastNamesMatchUsingAllComparisonChecks(String contactOneLastName, String contactTwoLastName){
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
+        return LastNameMatcher.getLastNameMatchResult(editedLastNameOne, editedLastNameTwo);
+    }
+
+    public static AnswerType doFirstAndLastNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
+        // ToDO: Discussion - What if the person got married/divorced and the last name was changed, but first name is still the same?? How to handle this case?
```

Comment:
I don't understand this case. Most people don't change their first names when they marry/divorce (??).

Reply:
Ah. I think so. If they retain their phone number, e-mail address, or address the contact match would come back as "possiblyRelated" and so we would still find them. I'd rather miss a match than declare a match when their isn't one because we might destroy/miss information. In other words, I'd rather add a "duplicate" contact than merge two different contacts.

Reply:
Okay, I will treat this as is (current default being return no).